src/index.rs

```diff
@@ -49,6 +51,14 @@ pub struct IndexEntry {
     #[serde(default)]
     pub mtime: i64,
 
+    /// Discretionary Access Control permissions (such as read/write/execute on unix)
+    #[serde(default)]
+    pub unix_mode: UnixMode,
```

Comment:
Actually, thinking about the case of old formats and backups from Windows: it seems like these should be `Option` rather than set to the default if missing. Then, restore should only attempt to set the permissions/owner if they are set.

Reply:
UnixMode is now UnixMode(Option<u32>), and restore cannot invoke fs::set_permissions if UnixMode is None, because it can't extract a Permissions struct from None.